packages/@expo/cli/src/start/server/__tests__/UrlCreator-test.ts

```diff
@@ -47,6 +47,11 @@ describe('constructDevClientUrl', () => {
     expect(createDefaultCreator().constructDevClientUrl({ scheme: 'http' })).toEqual(null);
     expect(createDefaultCreator().constructDevClientUrl({ scheme: 'https' })).toEqual(null);
   });
+  it(`returns null when protocol contains "_" characters`, () => {
+    expect(
+      createDefaultCreator().constructDevClientUrl({ scheme: 'dev.expo.invalid_node_protocol' })
```

Comment:
Can we confirm that the error message the user will get will **not** look something like "no scheme is defined"?

Reply:
The user won't get an error message when using `_`, it will instead just fallback to the Android "alternative" URL, which is `<identifier>/.MainActivity`